modules/apps/27-interchain-accounts/controller/keeper/keeper.go

```diff
@@ -0,0 +1,216 @@
+package keeper
+
+import (
+	"fmt"
+	"strings"
+
+	baseapp "github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	capabilitykeeper "github.com/cosmos/cosmos-sdk/x/capability/keeper"
+	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+	"github.com/tendermint/tendermint/libs/log"
+
+	controllertypes "github.com/cosmos/ibc-go/v2/modules/apps/27-interchain-accounts/controller/types"
+	"github.com/cosmos/ibc-go/v2/modules/apps/27-interchain-accounts/types"
+	channeltypes "github.com/cosmos/ibc-go/v2/modules/core/04-channel/types"
+	host "github.com/cosmos/ibc-go/v2/modules/core/24-host"
+)
+
+// Keeper defines the IBC interchain accounts controller keeper
+type Keeper struct {
+	storeKey sdk.StoreKey
+	cdc      codec.BinaryCodec
+
+	ics4Wrapper   types.ICS4Wrapper
+	channelKeeper types.ChannelKeeper
+	portKeeper    types.PortKeeper
+	accountKeeper types.AccountKeeper
+
+	scopedKeeper capabilitykeeper.ScopedKeeper
+
+	msgRouter *baseapp.MsgServiceRouter
+}
+
+// NewKeeper creates a new interchain accounts controller Keeper instance
+func NewKeeper(
```

Comment:
The module account is only required within the `host` submodule so I've removed the following check from `NewKeeper()` for the `controller`:

```
if addr := accountKeeper.GetModuleAddress(types.ModuleName); addr == nil {
	panic("the Interchain Accounts module account has not been set")
}
```